lib/upsert.rb

```diff
@@ -197,6 +202,7 @@ def initialize(connection, table_name, options = {})
     @merge_function_class = MergeFunction.const_get adapter
     @merge_function_cache = {}
     @assume_function_exists = options.fetch :assume_function_exists, true
+    @use_native_upsert = options.fetch :use_native_upsert, nil
```

Comment:
naming suggestion: call the user-facing option `disable_native` - e.g., by default, we will use native if possible, unless the user says `disable_native: true`.

`_upsert` is implied so you should omit.